src/io/state-file/schema.ts

```diff
@@ -252,30 +253,33 @@ const FrameOfReference: z.ZodType<FrameOfReference> = z.object({
   planeNormal: Vector3,
 });
 
-const Ruler: z.ZodType<Ruler> = z.object({
-  firstPoint: Vector3,
-  secondPoint: Vector3,
+const AnnotationTool = z.object({
   imageID: z.string(),
   frameOfReference: FrameOfReference,
   slice: z.number(),
   id: z.string(),
   name: z.string(),
   color: z.string(),
   label: z.string().optional(),
-});
+  labelName: z.string().optional(),
+}) satisfies z.ZodType<AnnotationTool>;
+
+const makeToolEntry = <T extends z.ZodRawShape>(tool: z.ZodObject<T>) =>
+  z.object({ tools: z.array(tool), labels: z.record(tool.partial()) });
 
-const Rectangle: z.ZodType<Optional<Rectangle, 'fillColor'>> = z.object({
+const Ruler = AnnotationTool.extend({
   firstPoint: Vector3,
   secondPoint: Vector3,
-  imageID: z.string(),
-  frameOfReference: FrameOfReference,
-  slice: z.number(),
+}) satisfies z.ZodType<Ruler>;
```

Comment:
`satisfies` seems to get the same type safety as the variable type declaration, while allowing me to `ZodObject.extend`.  Should I convert the rest of schema to use this pattern?

Reply:
Yep! This pattern looks good to me.